### ISSUE TITLE
shutdown_all_and_replace_node_test failed: unhandled exception: std::runtime_error (Cannot replace token -9195848202156986102 which does not exist!)

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#4361]
scylla version 57521747626903f0d37c75bbb702826c57fe71d4
scylla-dtest version scylladb/scylla-dtest@0d225442d4be6fb2bad830cb9cc0b9744208f831

Seen in [dtest-release/65/artifact/logs-release.2/1553317702547_replace_address_test.TestReplaceAddress.shutdown_all_and_replace_node_test/node4.log](http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest-release/65/artifact/logs-release.2/1553317702547_replace_address_test.TestReplaceAddress.shutdown_all_and_replace_node_test/node4.log):
```
ERROR 2019-03-23 06:59:30,191 [shard 0] seastar - Exiting on unhandled exception: std::runtime_error (Cannot replace token -9195848202156986102 which does not exist!)
```

Note that this dtest never passed.

**Comment by bhalevy** [comment]
@asias since you were the original author of this function please do the first analysis.

**Comment by slivne** [comment]
@asias ping

**Comment by asias** [comment]
The patch "storage_service: Allow replacing node when all nodes were shutdown" is on list and not merged yet.

**Comment by bhalevy** [comment]
Note, since scylladb/scylla-ccm@076d16239c8f64cc7636c46f993376b0bab4d038 the node failure is reported as [dtest-release/68/testReport/replace_address_test/TestReplaceAddress/shutdown_all_and_replace_node_test](http://jenkins.cloudius-systems.com:8080/view/master/job/scylla-master/job/dtest-release/68/testReport/replace_address_test/TestReplaceAddress/shutdown_all_and_replace_node_test/):
```
Stacktrace
  File "/usr/lib64/python2.7/unittest/case.py", line 367, in run
    testMethod()
  File "/jenkins/workspace/scylla-master/dtest-release@2/scylla-dtest/replace_address_test.py", line 128, in shutdown_all_and_replace_node_test
    node4.start(wait_for_binary_proto=True, replace_address=self.cluster.get_node_ip(3))
  File "/jenkins/workspace/scylla-master/dtest-release@2/scylla-ccm/ccmlib/scylla_node.py", line 380, in start
    wait_for_binary_proto)
  File "/jenkins/workspace/scylla-master/dtest-release@2/scylla-ccm/ccmlib/scylla_node.py", line 221, in _start_scylla
    self.wait_for_binary_interface(from_mark=self.mark, process=self._process_scylla)
  File "/jenkins/workspace/scylla-master/dtest-release@2/scylla-ccm/ccmlib/node.py", line 477, in wait_for_binary_interface
    self.watch_log_for("Starting listening for CQL clients", **kwargs)
  File "/jenkins/workspace/scylla-master/dtest-release@2/scylla-ccm/ccmlib/node.py", line 443, in watch_log_for
    raise RuntimeError("The process is dead, returncode={}".format(process.returncode))

The process is dead, returncode=1
```

**Comment by asias** [comment]
This is not a regression and we do not have agreement on the fix, so maybe 3.2?

**Comment by asias** [comment]
Closing. It is duplication of https://github.com/scylladb/scylla/issues/4325.
